### DISCUSSION TITLE
Device Parts Table

### DISCUSSION THREAD

**Discussion by hassankhan** [ludwig-v/wireless-carplay-dongle-reverse-engineering#61] [Show and tell]
| Device Name | CPU | RAM | Flash | Bluetooth/WiFi |
|---|---|---|---|---|
| **Carlinkit** | NXP/Freescale i.MX 6UltraLite | Micron/SK Hynix 1Gb(64x16) | Macronix 25L12835F (16MB) | Realtek RTL8822BS |
| **Carlinkit V2** | NXP/Freescale i.MX 6UltraLite | Micron/SK Hynix 1Gb(64x16) | Macronix 25L12835F (16MB) | Realtek RTL8822BS |
| **CPlay2Air** | NXP/Freescale i.MX 6UltraLite | Micron/SK Hynix 1Gb(64x16) | Macronix 25L12835F (16MB) | Realtek RTL8822BS |

**Comment by CiNcH83** [comment]
The CPU is just wrong. It is in fact an NXP/Freescale i.MX 6UltraLite. Actually all of those dongles use pretty much the same HW. Carlinkit V1 and CPLAY2air are exactly the same. Only difference according to @ludwig-v is that V2 uses active antennas for wireless.

RAM is 1Gb(64x16) (either Micron or SK hynix).

**Reply by hassankhan** [reply]
Ahh thanks for clarifying @CiNcH83; I mistakenly thought Carlinkit V2 and CPlay2Air were the same (i.e. with active antennas). I'll update the table now 👍

**Reply by CiNcH83** [reply]
BTW, I don't know why @ludwig-v still mentions the Microchip AT91SAM9260 in his README. It is not the chip in use for two reasons:

- Atmel/Microchip AT91SAM9260 has no DDR3 controller
- Atmel/Microchip AT91SAM9260 has no SDIO for RTL8822BS
- Linux syslog clearly states that the SoC is an i.MX6 (which makes a lot of sense as NXP/Freescale provide MFi development kits for Apple audio accessories)

It looks like somebody wants to obfuscate which chip is really being used.

**Reply by crazyyi** [reply]
Hi, some of you have a lot of knowledge about the HW. Do you think it's possible to form a team to build a prototype instead of DIY and updating firmwares for these C* products? It seems to me if the prototype can do the job of the dongles it would probably be able to self-funded and kickstart something bigger.

**Reply by iDoka** [reply]
I'm someone from pure HW world.
But I think task from your idea should targeted to FW guys.
Specs for Carplay & Airplay are describes a lot SW-stack (if somebody gonna to implement it from scratch).

**Reply by crazyyi** [reply]
> I'm someone from pure HW world.
> But I think task from your idea should targeted to FW guys.
> Specs for Carplay & Airplay are describes a lot SW-stack (if somebody gonna to implement it from scratch).

I know. I am a pure SW guy so the model names of these HW components mean little to me. But I believe there are some people who have knowledge about implementing a low level Carplay protocol onto the HW (maybe by hacking or just use the API provided by Apple, I am not very sure). Of course a lot of testing and debugging are needed.
